tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -408,6 +410,75 @@ def test_creating_user_with_existing_name_fails():
             zen_store.delete_user(user.id)
 
 
+def test_creating_users_in_parallel_do_not_duplicate_fails(
```

Comment:
these should have a `with pytest.raises(EntityExistsError):`, I guess?

Reply:
Not always - the first time it shall pass, but I cannot guarantee in which thread it will happen actually.